typings/index.d.ts

```diff
@@ -0,0 +1,1909 @@
+    onEnter: TransitionCallback;
+    onEntering: TransitionCallback;
+    onEntered: TransitionCallback;
+    onExit: TransitionCallback;
+    onExiting: TransitionCallback;
+    onExited: TransitionCallback;
+  };
+
+  export interface TransitionProps extends Partial<TransitionHandlers> {
+    children?: React.ReactElement<any>;
+    className?: string;
+    enteredClassName?: string;
+    enteringClassName?: string;
+    exitedClassName?: string;
+    exitingClassName?: string;
+    in?: boolean;
+    onRequestTimeout?: TransitionRequestTimeout;
+    timeout?: number;
+    transitionAppear?: boolean;
+    unmountOnExit?: boolean;
+  }
+
+  export default class Transition extends React.Component<TransitionProps> {}
+}
+
+/* ============================================= */
+/*                                               */
+/*                     STYLES                    */
+/*                                               */
+/* ============================================= */
+
+declare module 'material-ui/styles' {
+  export {
+    default as MuiThemeProvider,
+  } from 'material-ui/styles/MuiThemeProvider';
+  export { default as createBreakpoints } from 'material-ui/styles/breakpoints';
+  export { default as createMuiTheme } from 'material-ui/styles/theme';
+  export { default as createPalette } from 'material-ui/styles/palette';
+  export {
+    default as createStyleSheet,
+  } from 'material-ui/styles/createStyleSheet';
+  export { default as createTypography } from 'material-ui/styles/typography';
+  export { default as withStyles } from 'material-ui/styles/withStyles';
+  export { default as withTheme } from 'material-ui/styles/withTheme';
+}
+
+declare module 'material-ui/styles/MuiThemeProvider' {
+  import { Theme } from 'material-ui/styles/theme';
+
+  export interface MuiThemeProviderProps {
+    theme?: Theme<any>;
+    sheetsManager?: Object;
+    children: React.ReactNode;
+  }
+
+  export default class MuiThemeProvider extends React.Component<
+    MuiThemeProviderProps
+  > {}
+}
+
+declare module 'material-ui/styles/breakpoints' {
+  export type Breakpoint = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
+  export type BreakpointMap = { [key in Breakpoint]: number };
+  export const keys: Breakpoint[];
+
+  export interface BreakpointsOptions {
+    breakpoints: BreakpointMap;
+    unit: string;
+    step: number;
+  }
+
+  export interface Breakpoints {
+    keys: typeof keys;
+    values: BreakpointMap;
+    up: (key: Breakpoint) => string;
+    down: (key: Breakpoint) => string;
+    between: (start: Breakpoint, end: Breakpoint) => string;
+    only: (key: Breakpoint) => string;
+    getWidth: (key: Breakpoint) => number;
+  }
+
+  function createBreakpoints(
+    options?: Partial<BreakpointsOptions>
+  ): Breakpoints;
+
+  export default createBreakpoints;
+}
+
+declare module 'material-ui/styles/colorManipulator' {
+  export type ColorFormat = 'rgb' | 'rgba' | 'hsl' | 'hsla';
+  export type ColorObject = {
```

Comment:
This could be a disjoint type:

```ts
export type ColorObject = 
  | { type: 'rgb' | 'hsl'; color: [number, number, number] }
  | { type: 'rgba' | 'hsla'; color: [number, number, number, number] }
```